packages/components/src/range-control/types.ts

```diff
@@ -0,0 +1,326 @@
+		/**
+		 * Callback for when mouse moves within the `RangeControl`.
+		 *
+		 * @default () => void
+		 */
+		onMouseMove?: MouseEventHandler< HTMLInputElement >;
+		/**
+		 * CSS color string to customize the rail element's background.
+		 */
+		railColor?: CSSProperties[ 'color' ];
+		/**
+		 * A way to customize the rendered UI of the value.
+		 *
+		 * @default ( value ) => value
+		 */
+		renderTooltipContent?: (
+			value?: ControlledRangeValue
+		) => string | number | null | undefined;
+		/**
+		 * The value to revert to if the Reset button is clicked (enabled by
+		 * `allowReset`)
+		 */
+		resetFallbackValue?: number;
+		/**
+		 * Define if separator line under/above control row should be disabled
+		 * or full width. By default it is placed below excluding underline the
+		 * control icon.
+		 */
+		separatorType?: 'none' | 'fullWidth' | 'topFullWidth';
+		/**
+		 * Passed as a prop to the `NumberControl` component and is only
+		 * applicable if `withInputField` and `isShiftStepEnabled` are both true
+		 * and while the number input has focus. Acts as a multiplier of `step`.
+		 *
+		 * @default 10
+		 */
+		shiftStep?: number;
+		/**
+		 * Forcing the Tooltip UI to show or hide. This is overridden to `false`
+		 * when `step` is set to the special string value `any`.
+		 */
+		showTooltip?: boolean;
+		/**
+		 * CSS color string to customize the track element's background.
+		 */
+		trackColor?: CSSProperties[ 'color' ];
+		/**
+		 * Define if the value selection should present a stepper control or a
+		 * slider control in the bottom sheet on mobile. To use the stepper set
+		 * the type value as `stepper`. Defaults to slider if no option is
+		 * provided.
+		 */
+		type?: 'stepper';
+		/**
+		 * Determines if the `input` number field will render next to the
+		 * RangeControl. This is overridden to `false` when `step` is set to the
+		 * special string value `any`.
+		 *
+		 * @default true
+		 */
+		withInputField?: boolean;
+	};
+
+export type RailProps = MarksProps & {
+	railColor?: CSSProperties[ 'color' ];
+};
+
+export type InputRangeProps = {
+	describedBy?: string;
+	label?: string;
+	onHideTooltip?: () => void;
+	onMouseLeave?: MouseEventHandler< HTMLInputElement >;
+	onMouseMove?: MouseEventHandler< HTMLInputElement >;
+	onShowTooltip?: () => void;
+	value?: number | '';
+};
+
+export type WrapperProps = {
+	color?: CSSProperties[ 'color' ];
+	marks?: RangeMarks;
+};
+
+export type ThumbProps = {
+	isFocused?: boolean;
+	disabled?: boolean;
+};
+
+export type TooltipProps = {
+	show?: boolean;
+	position?: string;
+	inputRef?: MutableRefObject< HTMLElement | undefined >;
+	tooltipPosition?: string;
+	value?: ControlledRangeValue;
+	renderTooltipContent?: (
+		value?: ControlledRangeValue
+	) => string | number | null | undefined;
+	zIndex?: number;
+};
+
+export type TrackProps = {
+	disabled: boolean;
+	trackColor: CSSProperties[ 'color' ];
+};
+
+export type useControlledRangeValueArgs = {
```

Comment:
These should start with uppercase, no?

```suggestion
export type UseControlledRangeValueArgs = {
```

Reply:
Thanks, I've fixed up the casing for the custom hook types.